helios-services/src/main/java/com/spotify/helios/servicescommon/GooglePubSubProvider.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.servicescommon;
+
+import com.google.cloud.pubsub.PubSub;
+import com.google.cloud.pubsub.PubSubOptions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+
+public class GooglePubSubProvider {
+  private static final Logger log = LoggerFactory.getLogger(GooglePubSubProvider.class);
+
+  private final List<String> pubsubPrefixes;
+
+  public GooglePubSubProvider(final List<String> pubsubPrefixes) {
+    this.pubsubPrefixes = pubsubPrefixes;
+  }
+
+  public List<GooglePubSubSender> senders() {
```

Comment:
added some tests.
(`PubSubOptions.getService()` starts a real pubsub service straight away, so did a `senders(Supplier<PubSub> pubsubSupplier)` to test it)